src/xcode/ENA/ENA/Source/Scenes/ContactDiary/DiaryService.swift

```diff
@@ -55,97 +72,117 @@ class MockDiaryStore: DiaryStoring {
 	var diaryDaysPublisher = CurrentValueSubject<[DiaryDay], Never>([])
 
 	@discardableResult
-	func addContactPerson(name: String) -> Int {
+	func addContactPerson(name: String) -> DiaryStoringResult {
 		let id = contactPersons.map { $0.id }.max() ?? -1 + 1
 		contactPersons.append(DiaryContactPerson(id: id, name: name))
 
 		updateDays()
 
-		return id
+		return .success(id)
 	}
 
 	@discardableResult
-	func addLocation(name: String) -> Int {
+	func addLocation(name: String) -> DiaryStoringResult {
 		let id = locations.map { $0.id }.max() ?? -1 + 1
 		locations.append(DiaryLocation(id: id, name: name))
 
 		updateDays()
 
-		return id
+		return .success(id)
 	}
 
 	@discardableResult
-	func addContactPersonEncounter(contactPersonId: Int, date: String) -> Int {
+	func addContactPersonEncounter(contactPersonId: Int64, date: String) -> DiaryStoringResult {
 		let id = contactPersonEncounters.map { $0.id }.max() ?? -1 + 1
 		contactPersonEncounters.append(ContactPersonEncounter(id: id, date: date, contactPersonId: contactPersonId))
 
 		updateDays()
 
-		return id
+		return .success(id)
 	}
 
 	@discardableResult
-	func addLocationVisit(locationId: Int, date: String) -> Int {
+	func addLocationVisit(locationId: Int64, date: String) -> DiaryStoringResult {
 		let id = locationVisits.map { $0.id }.max() ?? -1 + 1
 		locationVisits.append(LocationVisit(id: id, date: date, locationId: locationId))
 
 		updateDays()
 
-		return id
+		return .success(id)
 	}
 
-	func updateContactPerson(id: Int, name: String) {
-		guard let index = contactPersons.firstIndex(where: { $0.id == id }) else { return }
+	func updateContactPerson(id: Int64, name: String) -> DiaryStoringVoidResult {
+		guard let index = contactPersons.firstIndex(where: { $0.id == id }) else { return .success(()) }
 		contactPersons[index] = DiaryContactPerson(id: id, name: name)
 
 		updateDays()
+
+		return .success(())
```

Comment:
Why no `throw` - if there is even a possible `fail` result? Looks like this whole thing is synchronous. Also, `.success(())` is very ugly.

Reply:
I think we should use `Result` because its type safe and more flexible.
Don't see a great need to use throw.
For my try and catch blocks are "very ugly" und unsafe.

Reply:
Also as a caller of the API, you can just ignore the success result, just unwrap the error.